hoodie-client/src/main/java/com/uber/hoodie/index/HoodieIndex.java

```diff
@@ -131,4 +133,11 @@ protected HoodieIndex(HoodieWriteConfig config, JavaSparkContext jsc) {
         }
         throw new HoodieIndexException("Index type unspecified, set " + config.getIndexType());
     }
+
+    /**
+     * Get latest file per partition, which is reused to reduce repeated lookup.
+     */
+    public Optional<LatestFileByPartitionInfo> getLatestFileByPartition() {
```

Comment:
how does this generalize to indexes which don't even lookup HDFS? e.g: external..

if not, then adding a new method to `HoodieIndex` is probably not the way to go